lib/internal/per_context/primordials.js

```diff
@@ -402,60 +402,30 @@ const makeSafe = (unsafe, safe) => {
   ObjectFreeze(safe);
   return safe;
 };
+
 primordials.makeSafe = makeSafe;
 
 // Subclass the constructors because we need to use their prototype
 // methods later.
 // Defining the `constructor` is necessary here to avoid the default
 // constructor which uses the user-mutable `%ArrayIteratorPrototype%.next`.
-primordials.SafeMap = makeSafe(
-  Map,
-  class SafeMap extends Map {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-primordials.SafeWeakMap = makeSafe(
-  WeakMap,
-  class SafeWeakMap extends WeakMap {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-
-primordials.SafeSet = makeSafe(
-  Set,
-  class SafeSet extends Set {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-primordials.SafeWeakSet = makeSafe(
-  WeakSet,
-  class SafeWeakSet extends WeakSet {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  },
-);
-
-primordials.SafeFinalizationRegistry = makeSafe(
-  FinalizationRegistry,
-  class SafeFinalizationRegistry extends FinalizationRegistry {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(cleanupCallback) { super(cleanupCallback); }
-  },
-);
-primordials.SafeWeakRef = makeSafe(
-  WeakRef,
-  class SafeWeakRef extends WeakRef {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(target) { super(target); }
-  },
-);
+const safeClasses = {
+  __proto__: null,
+  SafeMap: Map,
+  SafeWeakMap: WeakMap,
+  SafeSet: Set,
+  SafeWeakSet: WeakSet,
+  SafeFinalizationRegistry: FinalizationRegistry,
+  SafeWeakRef: WeakRef,
+  SafePromise: Promise
+};
 
-const SafePromise = makeSafe(
-  Promise,
-  class SafePromise extends Promise {
+for (const [safeName, unsafeClass] of ObjectEntries(safeClasses)) {
+  primordials[safeName] = makeSafe(unsafeClass, class extends unsafeClass {
     // eslint-disable-next-line no-useless-constructor
-    constructor(executor) { super(executor); }
-  },
-);
+    constructor(...args) { super(...args); }
```

Comment:
Using spread operator on an array relies on globally mutable methods (`Array.prototype[Symbol.iterator]`, `%ArrayIteratorPrototype%.next`). However, the version of V8 we use on `main` have already implemented https://github.com/tc39/ecma262/pull/2216, so we could simply remove the constructor declaration, and add `dont-land` label so this change doesn't get backported to current and LTS release lines.
```suggestion
```

Reply:
Okay, sounds good. Do I need to add any form of label, or can I just merge your commit into my PR?

Reply:
The labels are a GitHub thing, we probably can't add them yourself. You can find the docs about them, but you don't have to worry about it, it's specific to the way Node.js manages release lines.